build-system/app-index/components/main.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* eslint-disable no-unused-vars */
+
+import {Component, h, render} from 'preact';
+import {ExamplesDocumentModeSelect} from './examples-document-mode-select';
+import {ProxyForm} from './proxy-form';
+
+
+class Main extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = JSON.parse(document.querySelector('#initial-state').innerHTML);
```

Comment:
Since this is a developer view, can we just place the initial state in a `HTMLScriptElement` and leverage it directly?